src/base/emu-i386/kvm.c

```diff
@@ -158,6 +158,76 @@ static struct kvm_userspace_memory_region maps[MAXSLOT];
 
 static int init_kvm_vcpu(void);
 
+#if !defined(DISABLE_SYSTEM_WA) || !defined(KVM_CAP_IMMEDIATE_EXIT)
+
+/* compat functions for older complex method of immediate exit
+   using a signal that is blocked outside KVM_RUN but not blocked inside it */
+
+#include <pthread.h>
+
+#ifndef KVM_CAP_IMMEDIATE_EXIT
+#define KVM_CAP_IMMEDIATE_EXIT 136
+#define immediate_exit padding1[0]
+#endif
+
+#define KVM_IMMEDIATE_EXIT_SIG (SIGRTMIN + 1)
+static int kvm_cap_immediate_exit = -1;
+
+static void kvm_compat_set_immediate_exit(int set)
+{
+  if (kvm_cap_immediate_exit == -1) { // setup
+    struct sigaction sa;
+    sigset_t set;
+    struct kvm_signal_mask *mask;
+
+    kvm_cap_immediate_exit = ioctl(kvmfd, KVM_CHECK_EXTENSION, KVM_CAP_IMMEDIATE_EXIT) > 0;
```

Comment:
This is likely wrong to do under
`#if !defined(DISABLE_SYSTEM_WA) || !defined(KVM_CAP_IMMEDIATE_EXIT)`
because of the second part of a condition.

Reply:
Ah, you re-define it, sorry.